sei-tendermint/libs/utils/mutex.go

```diff
@@ -83,6 +83,24 @@ type atomicWatch[T any] struct {
 	ptr atomic.Pointer[version[T]]
 }
 
+type AtomicSend[T any] struct{ atomicWatch[T] }
+
+func (w *AtomicSend[T]) Subscribe() AtomicRecv[T] {
+	return AtomicRecv[T]{&w.atomicWatch}
+}
+
+// NewAtomicWatch creates a new AtomicWatch with the given initial value.
+func NewAtomicSend[T any](value T) (w AtomicSend[T]) {
+	w.ptr.Store(newVersion(value))
+	// nolint:nakedret
+	return
```

Comment:
why do we want the naked return and the nolint?

Reply:
AtomicSend is (intentionally) a no-copy object, because it embeds atomic.Pointer. Naked return is the canonical way of initializing no-copy objects (`return w` would be already a copy of x).